### PR TITLE
Feature better search

### PR DESCRIPTION
**Pull request by Max-F-Helm**
latests commits from 15.09.21 14:00

### CODE REVIEW

File: test/indexing.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    $config['password']
+);
+
+function myLog(string $level, string $msg): void {
+    // NOP
+}
+
+it('lists all tables which should be indexed', fn() => listTables($db) == ['mks_city', 'mks_collection', 'mks_genre', 'mks_person', 'mks_publisher', 'mks_song', 'mks_source']);
+
+file_put_contents("log.json", json_encode(indexTable('mks_person', $db), JSON_PRETTY_PRINT | JSON_UNESCAPED_UNICODE));
```
Comment (sesogr) [inline test/indexing.php:24]: This should be removed

File: helpers/lib/index_db.inc.php [hunk 2]
```diff
@@ -209,7 +209,12 @@ function indexTable(string $table, PDO $conn){
     }
 }
 
-function addIndex(array $indexData, PDO $conn){
+/**
+ * writes the index-data (created by indexTable) into the db
+ * @param array $indexData array of the format [word => [song-id, ...], ...]
+ * @param PDO $conn
+ */
+function writeIndex(array $indexData, PDO $conn){
```
Comment (sesogr) [inline helpers/lib/index_db.inc.php:221]: Good choice renaming this!

File: helpers/index_db.php [hunk 3]
```diff
@@ -2,12 +2,16 @@
...
-}
+$Loggers->addLogger(new class implements LoggerSubscriber{
+    public function log(string $type, string $message){
+        echo "$type : $message \n";
+    }
+});
 
 function run(): void {
-    myLog('info', 're-creating index');
+    global $Loggers;
```
Comment (sesogr) [inline helpers/index_db.php:12]: Better stay away from global. This might cause a bunch of problems in the long run.

Comment (Max-F-Helm) [reply helpers/index_db.php:12]: so how should i refernce the logger instead?

**Pull request merged**